pkg/controllers/status/cluster_condition_cache.go

```diff
@@ -46,9 +48,16 @@ func (c *clusterConditionStore) thresholdAdjustedReadyCondition(cluster *cluster
 		}
 		c.update(cluster.Name, saved)
 	}
-	if observedReadyCondition.Status != metav1.ConditionTrue &&
-		curReadyCondition.Status == metav1.ConditionTrue &&
-		now.Before(saved.thresholdStartTime.Add(c.failureThreshold)) {
+
+	var threshold time.Duration
+	if isConditionTrue(observedReadyCondition) {
+		threshold = c.successThreshold
+	} else {
+		threshold = c.failureThreshold
+	}
+
+	if isConditionTrue(observedReadyCondition) != isConditionTrue(curReadyCondition) &&
```

Comment:
```suggestion
	if observedReadyCondition != curReadyCondition &&
```
I can't see what's the difference between this, and the unit test seems ok with the change I'm suggesting.

Reply:
Because the condition can be `True/False/Unknown`, here we should only distinguish "true" and "not true" I think

For example, if the condition is changed from `False` to `Unknown`, we should update to `Unknown` immediately rather than retaining `False`

Reply:
> For example, if the condition is changed from False to Unknown,

The condition should never change from `true`/`false` to `unknown`. Any other examples?